components/post_emoji/index.jsx

```diff
@@ -12,17 +12,20 @@ import {getEmojiMap} from 'selectors/emojis';
 import PostEmoji from './post_emoji.jsx';
 
 function mapStateToProps(state, ownProps) {
+    const emojiName = ownProps.name;
     const emojiMap = getEmojiMap(state);
-    const emoji = emojiMap.get(ownProps.name);
+    const emoji = emojiMap.get(emojiName);
 
     let imageUrl = '';
     let displayTextOnly = false;
-    if (emoji) {
+    if (state.entities.emojis.nonExistentEmoji.has(emojiName)) {
+        displayTextOnly = true;
+    } else if (emoji) {
         imageUrl = getEmojiImageUrl(emoji);
+    } else if (getConfig(state).EnableCustomEmoji !== 'true' || getCurrentUserId(state) === '') {
+        displayTextOnly = false;
     } else {
-        displayTextOnly = state.entities.emojis.nonExistentEmoji.has(ownProps.name) ||
-            getConfig(state).EnableCustomEmoji !== 'true' ||
-            getCurrentUserId(state) === '';
```

Comment:
This last bit on how to determine whether to display text for named emoji (usually for custom emoji only) is the source of the problem.
Old: For named emoji which is not existing, the `displayTextOnly` usually returns true, mainly because `state.entities.emojis.nonExistentEmoji` doesn't have any entry yet.  Then the client request for that named emoji and if it responds longer, the UI will show a blank state until it resolves and fills `state.entities.emojis.nonExistentEmoji`.

Now: The `emojiMap` is already checking existence in custom emojis and so additional check whether the named emoji exists or not is not really needed.
On first render, it will show up as text unless it's included in the custom emojis or other check on config or current user state.